File: nim/collatz.nim
import math

proc collatz(seed: int): int =
    var steps = 0
    var tempSeed = seed
    while tempSeed > 1:
        if tempSeed mod 2 == 0:
            tempSeed = tempSeed div 2
        else:
            tempSeed = tempSeed * 3 + 1
        inc(steps)
    return steps

proc main() =
    var collatzResult = 0
    for seed in 1..1_000_000:
        collatzResult += collatz(seed)
    
    echo "steps ", collatzResult

main()

File: php/collatz.php
<?php

function collatz(int $seed): int {
    $steps = 0;
    while ($seed > 1) {
        while ($seed % 2 === 0) {
            $steps++;
            $seed /= 2;
        }
        if ($seed > 1) {
            $steps++;
            $seed = $seed * 3 + 1;
        }
    }
    return $steps;
}

function main() {
    $collatzResult = 0;
    for ($seed = 1; $seed <= 1000000; $seed++) {
        $collatzResult += collatz($seed);
    }
    printf("steps %d\n", $collatzResult);
}

main();
